### PR TITLE
Add API call to get a team by its name

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Add API call to get a team by its name

#### Ticket Link
Jira: https://mattermost.atlassian.net/browse/PLT-3762
GH: https://github.com/mattermost/platform/issues/4516

#### Checklist
- [X] Added or updated unit tests (required for all new features)
- [X] Added API documentation (required for all new APIs): https://github.com/mattermost/mattermost-api-reference/pull/50
- [X] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by cpanato** [comment]
Sure.  I forgot that my fault :/

File: webapp/actions/team_actions.jsx [hunk 1]
```diff
@@ -75,3 +75,21 @@ export function removeUserFromTeam(teamId, userId, success, error) {
         }
     );
 }
+
+export function getTeamByName(teamName, success, error) {
+    Client.removeUserFromTeam(
```
Comment (jwilander) [inline webapp/actions/team_actions.jsx]: I think this is a mistake, right? Also don't need to add an action if it's not being used anywhere

Comment (cpanato) [inline webapp/actions/team_actions.jsx]: you rigth! i was creating in everywhere :)

File: api/team.go [hunk 2]
```diff
@@ -708,6 +709,19 @@ func findTeamByName(c *Context, w http.ResponseWriter, r *http.Request) {
...
 }
 
+func getTeamByName(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	teamname := params["team_name"]
+
+	if result := <-Srv.Store.Team().GetByName(teamname); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
```
Comment (jwilander) [inline api/team.go]: Also once you have the team I think you need to do a permissions check like this:

```
if team.Type != model.TEAM_OPEN && c.Session.GetTeamByTeamId(team.Id) == nil {
	if !HasPermissionToContext(c, model.PERMISSION_MANAGE_SYSTEM) {
		return
	}
}
```

Comment (jwilander) [inline api/team.go]: Can you also add a case for this in the unit test? Essentially we want to make sure that if the team isn't open, then a user not on the team can't get it by name unless they are the system administrator

Comment (cpanato) [inline api/team.go]: sure!

**Comment by hmhealey** [comment]
Thanks @cpanato. The code looks good, but it has merge conflicts that need to be fixed

**Comment by hmhealey** [comment]
Thanks!

**Pull request merged**